Source/Global/global.cpp

```diff
@@ -22,9 +22,10 @@ enum class singleton_access_mode
     cleanup
 };
 
-std::shared_ptr<http_singleton> singleton_access(
+HRESULT singleton_access(
     _In_ singleton_access_mode mode,
-    _In_opt_ PerformEnv createArgs = {}
+    _Out_ std::shared_ptr<http_singleton>& singleton,
```

Comment:
usually `_Out_` params go last, but if you are really set on having createArg be optional, meh